terraform/test-fixtures/apply-destroy-nested-module-with-attrs/middle/bottom/bottom.tf

```diff
@@ -1 +1,5 @@
-variable "bottom_param" {}
+variable bottom_param {}
+
+resource "null_resource" "bottom" {
+  value = "${var.bottom_param}"
```

Comment:
Good question! In this test context, there is not an actual provider present. We [wire in a fake provider](https://github.com/hashicorp/terraform/blob/master/terraform/context_apply_test.go#L4816) and give it whatever name we want.

If you're actually asking about the proper `null_resource` behavior, it only supports `triggers` which is an arbitrary map of values that when changed will cause its provisioners to be re-run.

Hope his helps to clarify!